bosh/releases/pre_render_scripts/api/routing-api/bpm/patch_bpm.sh

```diff
@@ -5,15 +5,18 @@ set -o errexit -o nounset
 target="/var/vcap/all-releases/jobs-src/routing/routing-api/templates/bpm.yml.erb"
 
 PATCH=$(cat <<'EOT'
-@@ -11,7 +11,7 @@
+@@ -11,8 +11,10 @@
      - -timeFormat
      - rfc3339
      - -ip
 -    - <%= spec.ip %>
```

Comment:
Couldn't we just provide the Pod IP as `spec.ip` and not need a patch file at all?

Reply:
No, because `spec.ip` is not available during the bpm rendering, which happens on a pod before the `api` pod is created.

Reply:
Note that the statefulsets depend on the bpm rendering...

Reply:
Understood. However, this is a patch we cannot get rid of, right? That's pretty unfortunate.

Couldn't we parse the BPM for `<%= spec.ip %>` (match would have to be more fuzzy I suppose to catch all cases) and replace it by `POD_IP`? I pretty much mean exactly what you are already doing in this PR, we should just not require patching the bpm.yml file.

Reply:
I think we can get rid of this patch in the future by opening a PR on routing-release adding a property to control if it uses `spec.ip` or `$(POD_IP)`. I think this is more clear than cf-operator doing some magic on it.

Reply:
Mmh, why should the Bosh release know about POD_IP? And the "magic" would be pretty much in line with what the cf-operator does, "playing" Bosh director, wouldn't it?

Reply:
`$(POD_IP)` works here because it is a BPM argument that becomes a container argument in the k8s pod. It's not valid if `spec.ip` is used to set a BPM environment variable, which I believe I saw in the past. So, cf-operator would need to really do some magic and I don't think this kind of magic is cf-operator's responsibility. There are other releases that support more k8s specifics, and my suggestion aligns with it. Also, it becomes very clear and less error-prone.